samtranslator/metrics/metrics.py

```diff
@@ -0,0 +1,156 @@
+
+
+class DummyMetricsPublisher(MetricsPublisher):
+    def __init__(self):
+        MetricsPublisher.__init__(self)
+
+    def publish(self, namespace, metrics):
+        """Method to actually publish them metric"""
+        LOG.info("Dummy publisher ignoring {} metrices".format(len(metrics)))
+
+
+class Unit:
+    Seconds = "Seconds"
+    Microseconds = "Microseconds"
+    Milliseconds = "Milliseconds"
+    Bytes = "Bytes"
+    Kilobytes = "Kilobytes"
+    Megabytes = "Megabytes"
+    Bits = "Bits"
+    Kilobits = "Kilobits"
+    Megabits = "Megabits"
+    Percent = "Percent"
+    Count = "Count"
+
+
+class MetricDatum:
+    """
+    Class to hold Metric data.
+    """
+
+    def __init__(self, name, value, unit, dimensions=None):
+        """
+        Constructor
+
+        :param name: metric name
+        :param value: value of metric
+        :param unit: unit of metric (try using values from Unit class)
+        :param dimensions: array of dimensions applied to the metric
+        """
+        self.name = name
+        self.value = value
+        self.unit = unit
+        self.dimensions = dimensions if dimensions else []
+
+    def get_metric_data(self):
+        return {"MetricName": self.name, "Value": self.value, "Unit": self.unit, "Dimensions": self.dimensions}
+
+
+class Metrics:
+    def __init__(self, namespace="ServerlessTransform", metrics_publisher=None):
+        """
+        Constructor
+
+        :param namespace: namespace under which all metrics will be published
+        :param metrics_publisher: publisher to publish all metrics
+        """
+        self.metrics_publisher = metrics_publisher if metrics_publisher else DummyMetricsPublisher()
+        self.metrics_cache = []
+        self.namespace = namespace
+
+    def __del__(self):
+        if len(self.metrics_cache) > 0:
+            # attempting to publish if user forgot to call publish in code
+            LOG.warn("There are unpublished metrics. Please make sure you call publish after you record all metrics.")
+            self.publish()
+
+    def _record_metric(self, name, value, unit, dimensions=[]):
+        """
+        Create and save metric objects to an array.
+
+        :param name: metric name
+        :param value: value of metric
+        :param unit: unit of metric (try using values from Unit class)
+        :param dimensions: array of dimensions applied to the metric
+        """
+        self.metrics_cache.append(MetricDatum(name, value, unit, dimensions))
+
+    def record_count(self, name, value, unit=Unit.Count, dimensions=[]):
+        """
+        Create metric with unit Count.
+
+        :param name: metric name
+        :param value: value of metric
+        :param unit: unit of metric (try using values from Unit class)
+        :param dimensions: array of dimensions applied to the metric
+        """
+        self._record_metric(name, value, unit, dimensions)
+
+    def record_latency(self, name, value, unit=Unit.Milliseconds, dimensions=[]):
```

Comment:
Given that these methods are specialized, can we make them easier by removing `unit` and `dimensions` parameters?

Reply:
I'll remove `unit` we kind of need `dimensions`